src/handlers/collect-linked-pulls.ts

```diff
@@ -1,78 +1,19 @@
 import { parseIssueUrl } from "../helpers/github-url";
 import { Context } from "../types/context";
-import { GitHubLinkEvent, GitHubTimelineEvent, isGitHubLinkEvent } from "../types/github-types";
+import { IssuesSearch } from "../types/github-types";
 
 export type IssueParams = ReturnType<typeof parseIssueUrl>;
 
-export async function collectLinkedPullRequests(context: Context, issue: IssueParams) {
-  const onlyPullRequests = await collectLinkedPulls(context, issue);
-  return onlyPullRequests.filter((event) => {
-    if (!event.source.issue.body) {
-      return false;
-    }
-    // Matches all keywords according to the docs:
-    // https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-    // Works on multiple linked issues, and matches #<number> or URL patterns
-    const linkedIssueRegex =
-      /\b(?:Close(?:s|d)?|Fix(?:es|ed)?|Resolve(?:s|d)?):?\s+(?:#(\d+)|https?:\/\/(?:www\.)?github\.com\/(?:[^/\s]+\/[^/\s]+\/(?:issues|pull)\/(\d+)))\b/gi;
-    const linkedPrUrls = event.source.issue.body.match(linkedIssueRegex);
-    if (!linkedPrUrls) {
-      return false;
-    }
-    let isClosingPr = false;
-    for (let i = 0; i < linkedPrUrls.length && !isClosingPr; ++i) {
-      const idx = linkedPrUrls[i].indexOf("#");
-      if (idx !== -1) {
-        isClosingPr = Number(linkedPrUrls[i].slice(idx + 1)) === issue.issue_number;
-      } else {
-        const url = linkedPrUrls[i].match(/https.+/)?.[0];
-        if (url) {
-          const linkedRepo = parseIssueUrl(url);
-          isClosingPr = linkedRepo.issue_number === issue.issue_number && linkedRepo.repo === issue.repo && linkedRepo.owner === issue.owner;
-        }
-      }
-    }
-    return isGitHubLinkEvent(event) && event.source.issue.pull_request?.merged_at === null && isClosingPr;
-  });
+function additionalBooleanFilters(issueNumber: number) {
+  return `linked:${issueNumber} in:body "closes #${issueNumber}" OR "closes #${issueNumber}" OR "fixes #${issueNumber}" OR "fix #${issueNumber}" OR "resolves #${issueNumber}"`;
```

Comment:
```typescript
["closes", "fixes", "fix", "resolves"].map(o => `${o} #${issueNumber}`).join(" OR ")
```
might be less error prone. Also you do not cover all the possible keywords, see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

Reply:
https://github.com/ubiquibot/user-activity-watcher/pull/15